### PR TITLE
feat: shelf changes for 7mode and cdot

### PR DESCRIPTION
**Pull request by rahulguptajss**

### CODE REVIEW

File: cmd/collectors/zapi/plugins/shelf/shelf.go [hunk 1]
```diff
@@ -149,8 +150,7 @@ func (my *Shelf) Run(data *matrix.Matrix) ([]*matrix.Matrix, error) {
 	if x := result.GetChildS("attributes-list"); x != nil {
 		shelves = x.GetChildren()
```
Comment (Hardikl) [inline cmd/collectors/zapi/plugins/shelf/shelf.go:151]: can we wrap other code in handlecmode() ?

File: cmd/collectors/zapi/plugins/shelf/shelf.go [hunk 2]
```diff
@@ -280,3 +280,149 @@ func (my *Shelf) Run(data *matrix.Matrix) ([]*matrix.Matrix, error) {
...
+					output = append(output, data1)
+				}
+			}
+		}
+	}
+
+	// second loop to populate numeric data
+
+	for _, channel := range channels {
+		channelName := channel.GetChildContentS("channel-name")
```
Comment (Hardikl) [inline cmd/collectors/zapi/plugins/shelf/shelf.go:380]: do we want to complete channel loop and again do iterate? or can we do 2 for loop of shelves in same channel loop?

File: cmd/collectors/zapi/plugins/shelf/shelf.go (hunk 1 above)
Comment (rahulguptajss) [reply cmd/collectors/zapi/plugins/shelf/shelf.go]: done

File: cmd/collectors/zapi/plugins/shelf/shelf.go (hunk 2 above)
Comment (rahulguptajss) [reply cmd/collectors/zapi/plugins/shelf/shelf.go]: done

**Review by Hardikl — DISMISSED** [review]

File: cmd/collectors/zapi/plugins/shelf/shelf.go [hunk 3]
```diff
@@ -201,7 +209,7 @@ func (my *Shelf) Run(data *matrix.Matrix) ([]*matrix.Matrix, error) {
 						instance, err := data1.NewInstance(instanceKey)
 
 						if err != nil {
-							my.Logger.Debug().Msgf("add (%s) instance: %v", attribute, err)
+							my.Logger.Error().Msgf("add (%s) instance: %v", attribute, err)
```
Comment (cgrinds) [inline cmd/collectors/zapi/plugins/shelf/shelf.go:212]: my.Logger.Error().Err(err).Str("attribute", attribute).Msg("Failed to add instance")

File: cmd/collectors/zapi/plugins/shelf/shelf.go [hunk 4]
```diff
@@ -233,50 +252,112 @@ func (my *Shelf) Run(data *matrix.Matrix) ([]*matrix.Matrix, error) {
...
 
-	for _, shelf := range shelves {
+func (my *Shelf) handle7Mode(result *node.Node) ([]*matrix.Matrix, error) {
+	var (
+		shelves  []*node.Node
+		channels []*node.Node
+	)
+	//fallback to 7mode
+	channels = result.SearchChildren([]string{"shelf-environ-channel-info"})
+	//shelves = result.SearchChildren([]string{"shelf-environ-channel-info", "shelf-environ-shelf-list", "shelf-environ-shelf-info"})
```
Comment (cgrinds) [inline cmd/collectors/zapi/plugins/shelf/shelf.go:265]: dead code that can be removed?

**Review by cgrinds — COMMENTED** [review]
nice job @rahulguptajss

**Review by Hardikl — APPROVED** [review]

**Pull request merged**